java/arcs/core/storage/Reference.kt

```diff
@@ -35,6 +35,28 @@ data class Reference(
 
     override suspend fun dereference(coroutineContext: CoroutineContext): RawEntity? =
         requireNotNull(dereferencer).dereference(this, coroutineContext)
+
+    /** Entity creation time (in millis). */
+    @Suppress("GoodTime") // use Instant
+    override var creationTimestamp: Long = RawEntity.UNINITIALIZED_TIMESTAMP
+        set(value) {
+            require(this.creationTimestamp == RawEntity.UNINITIALIZED_TIMESTAMP) {
+                "cannot override creationTimestamp $value"
+            }
+            @Suppress("GoodTime") // use Instant
+            field = value
+        }
+
+    /** Entity expiration time (in millis). */
+    @Suppress("GoodTime") // use Instant
```

Comment:
Would it be possible to actually use Instant rather than just suppressing this?

Reply:
it's not cross platform :(

Reply:
That's so sad... 😦